syncer/syncer.go

```diff
@@ -1404,16 +1438,37 @@ func (s *Syncer) syncDML(
 
 // Run starts running for sync, we should guarantee it can rerun when paused.
 func (s *Syncer) Run(ctx context.Context) (err error) {
-	tctx := s.tctx.WithContext(ctx)
+	runCtx, runCancel := context.WithCancel(context.Background())
+	tctx := s.tctx.WithContext(runCtx)
 
 	defer func() {
 		if s.done != nil {
 			close(s.done)
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
+		tctx.L().Info("received subtask's done")
+		if s.isTransactionEnd.Load() {
```

Comment:
why we need isTransactionEnd? can we just use waitComplete instead?

Reply:
This is to determine whether the current transaction is over